### DISCUSSION TITLE
Timeline Interpretation

### DISCUSSION THREAD

**Discussion by wabiloo** [avishaycohen/mpd-parser#26] [Ideas]
Hi,

I'm using your lib in a project. Thanks for building it (so I didn't have to).
However I'm in need of additional functionality, in particular for computing the presentation timeline.
Things like parsing iso timestamps into date times, iso durations into durations, computation of effective period start times (in particular for live) and durations (in particular for multi-period), of segment start and end, etc.

Have you considered adding this?
Or do you know of another module that would have built that (maybe on top of yours)?

I am happy to fork and contribute if you think that could be useful?

**Comment by avishaycohen** [comment]
I'll be happy to implement any suggestions. Just consider I so this in my
free time, so it might take a week or two depending on how complex the
demend.
Anyway, please open an issue and describe your feature request, preferably
with a manifest that shows the behavior.
In general I thought about making general transformers for certain data
types to take the strings and make them easier to work with, but haven't
prioritized it as I feel the project needs a more robust testing (as your
previous issue shows).

**Reply by wabiloo** [reply]
Hi again,
Given that I'm on a project at the moment, and quite active, and that I need quite a bit more functionality beyond just a parser, I decided to form and build my own.
It's at https://github.com/wabiloo/mpd-inspector.

I think I should contribute some of the changes, as they fix some issues with the MPD structure. I'll see if I can find an easy way to do that.

Thanks for the excellent starting point you gave me!

**Comment by avishaycohen** [comment]
@wabiloo please check the latest release (0.1.4). let me know what else you would like me to add.
